lib/EntryDownloader.js

```diff
@@ -13,13 +15,15 @@ var events = require("events");
 
 var EntryDownloader = (function () {
 
-    function EntryDownloader(entryId, hostname, port, applicationName, manifestTimeWindow, manifestMaxChunkCount) {
+    function EntryDownloader(entryId, hostname, port, applicationName, flavorsIds, manifestTimeWindow, manifestMaxChunkCount, flavorBase) {
         events.EventEmitter.call(this);
         this.entryId = entryId;
         this.manifestTimeWindow = manifestTimeWindow;
         this.manifestMaxChunkCount = manifestMaxChunkCount;
         this.flavorsDownloaders = [];
-        this.masterManifestGenerator = masterManifestGeneratorCreator(entryId, hostname, port, applicationName, logger);
+        this.manifestGenerator = (flavorBase) ? flavorsBasedManifest(entryId, hostname, port, applicationName, flavorsIds, logger) :
+            smilBasedManifest(entryId, hostname, port, applicationName, logger);
```

Comment:
1.use flavorBase on the configuration and not as a parameter to the EntryDownload
2. instead of 

 this.manifestGenerator = (flavorBase) ? flavorsBasedManifest(entryId, hostname, port, applicationName, flavorsIds, logger) :
 +            smilBasedManifest(entryId, hostname, port, applicationName, logger);


do

var manifestGeneratorFn= (config.get('manifestGenerator')) ? flavorsBasedManifest : smilBasedManifest;
 this.manifestGenerator =manifestGeneratorFn(entryId, hostname, port, applicationName, flavorsIds, logger) :